### PR TITLE
Add CRUD server routes for Boards. Add auth tests for Boards server route.

### PR DESCRIPTION
**Pull request by stevepkuo**
Also add middleware authorization checks to make sure user is authorized to access Boards server routes.

Updated seed dummy data and update profileModel.spec expectations. Update server.apicrud.spec to expect different result after hooking up with latest db helper functions.

Fix /profile route to return user profile.

### CODE REVIEW

File: server/middleware/auth.js [hunk 1]
```diff
@@ -9,6 +20,79 @@ module.exports.verify = (req, res, next) => {
...
+  }
+  var userId = req.user.id;
+
+  // return models.User.where({ id: userId }).fetch({withRelated:['memberOfBoards']})
+  // .then(function(user) {
+  //   return user.related('memberOfBoards');
+  // })
+
+  //see whether the boardid shows up under any of the users boards
+  return models.User.where({ id: userId }).fetch()
```
Comment (brendanusa) [inline server/middleware/auth.js]: everything up to .filter can be handled by getBoardsByUser

File: server/middleware/auth.js [hunk 2]
```diff
@@ -9,6 +20,79 @@ module.exports.verify = (req, res, next) => {
...
+    boardid = parseInt(req.params.id);
+  }
+  var userId = req.user.id;
+
+  // return models.User.where({ id: userId }).fetch({withRelated:['memberOfBoards']})
+  // .then(function(user) {
+  //   return user.related('memberOfBoards');
+  // })
+
+  //see whether the boardid shows up under any of the users boards
```
Comment (brendanusa) [inline server/middleware/auth.js]: everything below up to .filter can be handled by getBoardsByUser

File: server/controllers/boards.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      createdBoard = result;
+      return dbhelper.addUserToBoard(req.user.id, result.id);
+    })
+    .then(result => {
+      if (!result) {
+        throw 'could not register user for the board';
+      }
+      res.status(201).send(createdBoard);
+    })
+    .catch(err => {
```
Comment (brendanusa) [inline server/controllers/boards.js]: createBoard and addUserToBoard already check for duplicates

File: server/middleware/auth.js (hunk 2 above)
Comment (stevepkuo) [reply server/middleware/auth.js]: Changed to getBoardsByUser

File: server/controllers/boards.js (hunk 3 above)
Comment (stevepkuo) [reply server/controllers/boards.js]: I have removed it.

Comment (brendanusa) [reply server/controllers/boards.js]: Excellent. It is out of our lives forever

**Pull request merged**